sqlglot/generator.py

```diff
@@ -99,34 +99,39 @@ class Generator:
 
     STRUCT_DELIMITER = ("<", ">")
 
-    BEFORE_PROPERTIES = {
-        exp.FallbackProperty,
-        exp.WithJournalTableProperty,
-        exp.LogProperty,
-        exp.JournalProperty,
-        exp.AfterJournalProperty,
-        exp.ChecksumProperty,
-        exp.FreespaceProperty,
-        exp.MergeBlockRatioProperty,
-        exp.DataBlocksizeProperty,
-        exp.BlockCompressionProperty,
-        exp.IsolatedLoadingProperty,
-    }
-
-    ROOT_PROPERTIES = {
-        exp.ReturnsProperty,
-        exp.LanguageProperty,
-        exp.DistStyleProperty,
-        exp.DistKeyProperty,
-        exp.SortKeyProperty,
-        exp.LikeProperty,
-    }
-
-    WITH_PROPERTIES = {
-        exp.Property,
-        exp.FileFormatProperty,
-        exp.PartitionedByProperty,
-        exp.TableFormatProperty,
+    PROPERTIES_LOCATION = {
```

Comment:
If we want to keep this, then maybe it can be simplified by doing something like:

```python
PROPERTIES_LOCATION = {
    "pre_schema": {
        exp.AfterJournalPropery,
        ...
    },
    "post_schema_root": { ... }
}
```

Reply:
Or, as Toby mentioned, we can also use enum values as keys.

Reply:
that’s better

Reply:
let’s go with george’s solution

Reply:
That's how I originally implemented it, seems like it could go either way.

This is simpler in the processing code, a bit more complicated for dialects to override (although that may not be a real concern).

Reply:
i see, it's harder to override because then you have to get the inner.

ok, let's go with enums, the dialects should be easier to override

Reply:
Updated to use enums